R/read.R

```diff
@@ -28,6 +29,20 @@ orderly_read_r <- function(path) {
   names(dat) <- sub("^orderly_", "", names(dat))
 
   ret <- list()
+  if (length(dat$parameters) > 0) {
+    ## TODO: once things are working, check for no duplicate parameter
+    ## names across multiple calls; leaving this assertion in for
+    ## now. Reporting on that well suggests that we should be able to
+    ## record the line numbers when we pass off through the file;
+    ## that's not too hard to do really, I think we can do that with
+    ## getSrcref or similar.
+    ##
+    ## TODO: we should disallow use of the parameter function in any
+    ## deeper expression (at least in strict mode), worth searching
+    ## for that and setting up that check here
```

Comment:
Do you want to address any of these or leave them until later?

Reply:
Later, both of these are fairly complex and we'll hit similar issues in other parts I think